Sources/TuistCore/Graph/XCFrameworkNode.swift

```diff
@@ -0,0 +1,64 @@
+import Basic
+import Foundation
+import TuistSupport
+
+final class XCFrameworkNode: PrecompiledNode {
+    let libraries: [XCFrameworkInfoPlist.Library]
+
+    init(path: AbsolutePath, libraries:  [XCFrameworkInfoPlist.Library]) {
+        self.libraries = libraries
+        super.init(path: path)
+    }
+
+    static func parse(
```

Comment:
Models should not contain business logic because that complicates testing. Let's say we have the following scenario _(note that is pseudocode)_:

```swift
class MyClass {
  func method(graph: Graph) {
    let xcframeworks = graph.xcframeworks()
    let binaryPaths = xcframeworks.map(\.binaryPath))
    // Some other stuff
  }
}
```

We'd need to make the `XCFrameworkNode` subclassable to be able to override the `binaryPath` method and test different scenarios.

For this reason I removed the business logic from the precompiled library and framework models and extracted it into what I called medatata providers. [This](https://github.com/tuist/tuist/blob/master/Sources/TuistCore/Utils/LibraryMetadataProvider.swift) is the metadata provider for libraries, and [this one](https://github.com/tuist/tuist/blob/master/Sources/TuistCore/Utils/FrameworkMetadataProvider.swift) is the one for frameworks.

That being said, I'd update the parser to not do IO using the metadata provider _(i.e. remove side effects)_, and remove the `binaryPath` methods.

This change not only will make testing easier but also facilitate future optimizations because those side effects are well scoped.